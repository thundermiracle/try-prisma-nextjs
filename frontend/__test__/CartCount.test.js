import { shallow, mount } from "enzyme";
import toJSON from "enzyme-to-json";
import CartCount from "../components/CartCount";

describe("CartCount by snapshot", () => {
  it("renders", () => {
    shallow(<CartCount count={10} />);
  });

  it("snapshot", () => {
    const wrapper = shallow(<CartCount count={10} />);

    expect(toJSON(wrapper)).toMatchSnapshot();
  });

  it("update via props", () => {
    const wrapper = shallow(<CartCount count={15} />);
    expect(toJSON(wrapper)).toMatchSnapshot();
    wrapper.setProps({ count: 10 });
    expect(toJSON(wrapper)).toMatchSnapshot();
  });
});
