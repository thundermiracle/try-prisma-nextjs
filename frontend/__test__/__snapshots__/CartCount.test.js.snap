// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CartCount by snapshot snapshot 1`] = `
<CartCount__AnimationStyles>
  <TransitionGroup
    childFactory={[Function]}
    component="div"
  >
    <CSSTransition
      className="count"
      classNames="count"
      key="10"
      timeout={400}
      unmountOnExit={true}
    >
      <CartCount__Dot>
        10
      </CartCount__Dot>
    </CSSTransition>
  </TransitionGroup>
</CartCount__AnimationStyles>
`;

exports[`CartCount by snapshot update via props 1`] = `
<CartCount__AnimationStyles>
  <TransitionGroup
    childFactory={[Function]}
    component="div"
  >
    <CSSTransition
      className="count"
      classNames="count"
      key="15"
      timeout={400}
      unmountOnExit={true}
    >
      <CartCount__Dot>
        15
      </CartCount__Dot>
    </CSSTransition>
  </TransitionGroup>
</CartCount__AnimationStyles>
`;

exports[`CartCount by snapshot update via props 2`] = `
<CartCount__AnimationStyles>
  <TransitionGroup
    childFactory={[Function]}
    component="div"
  >
    <CSSTransition
      className="count"
      classNames="count"
      key="10"
      timeout={400}
      unmountOnExit={true}
    >
      <CartCount__Dot>
        10
      </CartCount__Dot>
    </CSSTransition>
  </TransitionGroup>
</CartCount__AnimationStyles>
`;
